### DISCUSSION TITLE
Multi-level dropdown menus

### DISCUSSION THREAD

**Discussion by kooriookami** [element-plus/element-plus#912] [Ideas]
Hey, I think the dropdown menu can be multi-leveled. It differs from cascading menus, which focus on selection and are often used in conjunction with input boxes. Multi-level dropdown menus focus on the menu itself, such as right-click menus. They also vary in styles, as shown in the figure below.

Dropdown menus of unequal height.
![image](https://user-images.githubusercontent.com/38392315/101591994-80446180-3a28-11eb-9bb7-ef03c0b04fd0.png)

Cascading menus require equal height.
![image](https://user-images.githubusercontent.com/38392315/101592406-2728fd80-3a29-11eb-90fc-260b6d5663b4.png)

You can refer to View UI.
http://v1.iviewui.com/components/dropdown

**Comment by jw-foss** [comment]
Seems to me that this is typical cascader situation,  also somehow like nav-menu situation. I dunno if it is worthy to implement.
If you want to have this on the local branch I think its legit to for this repo to create one, or just do some encapsulation with `el-select` in you project it will work, you can refer to `NavMenu` for more insights.

**Comment by troy351** [comment]
This is useful when implement actions, like context menu
